drivers/spi/spi_sam0.c

```diff
@@ -318,27 +335,40 @@ static void spi_sam0_fast_transceive(struct spi_config *config,
  * - Zero or more trailing RX only bufs
  * - Zero or more trailing TX only bufs
  */
-static bool spi_sam0_is_regular(const struct spi_buf *tx_bufs,
-				size_t tx_count, struct spi_buf *rx_bufs,
-				size_t rx_count)
+static bool spi_sam0_is_regular(const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
 {
+	const struct spi_buf *tx;
+	const struct spi_buf *rx;
+	size_t tx_count;
+	size_t rx_count;
+
+	if (!tx_bufs && !rx_bufs) {
+		return false;
```

Comment:
It seems that there is something wrong here. When both `tx_bufs` and `rx_bufs` are `NULL` the function returns immediately but when only one of them is `NULL`, this pointer is dereferenced below.